numba/tests/test_array_exprs.py

```diff
@@ -395,6 +397,8 @@ def test_issue_1264(self):
         expected = distance_matrix(x)
         actual = njit(distance_matrix)(x)
         np.testing.assert_array_almost_equal(expected, actual)
+        # Avoid sporadic failures in MemoryLeakMixin.tearDown()
+        gc.collect()
```

Comment:
As explained privately:
- there are places where we want to ensure no reference cycle was created
- `gc.collect` can be slow (50-100 ms may not be surprising, depending on the number of tracked objects) so I'd rather avoid calling it where it's pointless